### PR TITLE
Only show current and all shops for specific price type

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | https://github.com/PrestaShop/PrestaShop/issues/30120
| Type?             | refacto
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | https://github.com/PrestaShop/PrestaShop/issues/30120
| Related PRs       | 
| How to test?      | refer to https://github.com/PrestaShop/PrestaShop/issues/30120. Also Check that the list of specific prices in new product page only shows specific prices for current shop OR for All shops.
| Possible impacts? |

### CODE REVIEW

File: src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/ApplicableGroupsType.php [hunk 1]
```diff
@@ -125,18 +132,33 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
...
             ]);
         }
     }
 
+    /**
+     * @return array<string, int>
+     */
+    private function buildShopChoices(): array
+    {
+        $choices = [];
```
Comment (jolelievre) [inline src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/ApplicableGroupsType.php]: All stores should be put before the list of shops I think (even if it's not the initial default choice) so you can init you array like this:

```suggestion
        $choices = [
            $this->trans('All stores', 'Admin.Global') => 0,
        ];
```

**Review by jolelievre — COMMENTED** [review]
It's a good start but a few things are missing I believe:
- the list of specific prices must be filtered and only show elements assigned to current shop OR to all shops
- in the creation form the shop association must be prefilled with current shop

**Review by lartist — DISMISSED** [review]

**Comment by lartist** [comment]
I approved but mind the CI is red. Some fixes are needed.

**Review by jolelievre — DISMISSED** [review]
Thanks @zuk3975

**Review by lartist — DISMISSED** [review]

**Review by nicosomb — DISMISSED** [review]

**Review by MhiriFaten — DISMISSED** [review]
Hello @zuk3975 ,

I have checked the PR and the automated test are OK ! 
It is QA aprouved :heavy_check_mark: 

Thank you.

**Comment by jolelievre** [comment]
Closed/reopened the PR because sanity tests were stuck

**Review by jolelievre — APPROVED** [review]
Reapproved after rebase to unlock CI

**Review by nicosomb — APPROVED** [review]

**Comment by nicosomb** [comment]
Thank you @zuk3975

**Pull request merged**